src/zenml/logging/step_logging.py

```diff
@@ -0,0 +1,230 @@
+        # local path otherwise, set local_logging_file to a temporary file
+        if is_remote(self.logs_uri):
+            # We log to a temporary file first, because
+            # TimedRotatingFileHandler does not support writing
+            # to a remote file, but still needs a file to get going
+            local_logging_file = f".zenml_tmp_logs_{int(time.time())}"
+            self.local_temp_file = local_logging_file
+        else:
+            local_logging_file = self.logs_uri
+
+        super().__init__(
+            local_logging_file,
+            when="s",
+            interval=LOGS_HANDLER_INTERVAL_SECONDS,
+        )
+
+    def emit(self, record: LogRecord) -> None:
+        """Emits the log record.
+
+        Args:
+            record: Log record to emit.
+        """
+        msg = self.format(record)
+        self.buffer.write(msg + "\n")
+        self.message_count += 1
+
+        current_time = time.time()
+        time_elapsed = current_time - self.last_upload_time
+
+        if (
+            self.message_count >= self.max_messages
+            or time_elapsed >= self.interval
+        ):
+            self.flush()
+
+    def flush(self) -> None:
+        """Flushes the buffer to the artifact store."""
+        try:
+            if not self.disabled:
+                self.disabled = True
+                with fileio.open(self.logs_uri, mode="wb") as log_file:
+                    log_file.write(self.buffer.getvalue().encode("utf-8"))
+        except (OSError, IOError) as e:
+            # This exception can be raised if there are issues with the
+            # underlying system calls, such as reaching the maximum number of
+            # open files, permission issues, file corruption, or other
+            # I/O errors.
+            logger.error(f"Error while trying to write logs: {e}")
+        finally:
+            self.disabled = False
+
+        self.message_count = 0
+        self.last_upload_time = time.time()
+
+    def doRollover(self) -> None:
+        """Flushes the buffer and performs a rollover."""
+        self.flush()
+        super().doRollover()
+
+    def close(self) -> None:
+        """Tidy up any resources used by the handler."""
+        self.flush()
+        super().close()
+
+        try:
+            if not self.disabled:
+                self.disabled = True
+                if self.local_temp_file and fileio.exists(
+                    self.local_temp_file
+                ):
+                    fileio.remove(self.local_temp_file)
+        except (OSError, IOError) as e:
+            # This exception can be raised if there are issues with the
+            # underlying system calls, such as reaching the maximum number of
+            # open files, permission issues, file corruption, or other
+            # I/O errors.
+            logger.error(f"Error while close the logger: {e}")
+        finally:
+            self.disabled = False
```

Comment:
same here: the `if not self-disabled` check must be done outside the try-finally block

Reply:
Changed this one as well.